Makefile

```diff
@@ -0,0 +1,58 @@
+# 
+# Test and build project
+# 
+# This Makefile is primarily instructional; you can simply enter the Nix environment for
+# holochain-rust development (supplied by holo=nixpkgs; see pkgs.nix) via `nix-shell` and
+# perform Rust and Javascript project build and test procedures manually, if you prefer.
+#
+
+SHELL		= bash
+
+
+# External targets; Uses a nix-shell environment to obtain Holochain runtimes, run tests, etc.
+.PHONY: all
+all: 		nix-test
+
+
+# nix-test, nix-install, ...
+# 
+# Provides a nix-shell environment, and runs the desired Makefile target.  It is recommended that
+# you add `substituters = ...` and `trusted-public-keys = ...` to your nix.conf (see README.md), to
+# take advantage of cached Nix and Holo build assets.
+nix-%:
+	nix-shell --pure --run "make $*"
```

Comment:
This is a bad idea, please don't add this to Makefile. Users should enter `nix-shell`, rather than creating and destroying shell environment with every command. This can have all sorts of unintended consequences and bad user experience, ranging from services being started/stopped on every comamnd, to build times that are not adequate for iterative development. I've already stated that before in one of `happ-example` reviews.